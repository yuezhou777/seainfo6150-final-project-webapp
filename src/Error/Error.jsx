import React from 'react';
import { Link } from "react-router-dom";
import error from "../images/ErrorPage/error.jpg";
import NavBar from "../NavBar/NavBar";
import styles from './Error.module.css';

const Error = () => {
    return (
        <div className={styles.container}>
          <div>
            <NavBar/>
          </div>
          <div>
            <img src={error} className={styles.errorImg} alt="404 Not Found" />
          </div>
          <div>
            <Link to="/" className={styles.link}>Back To Main</Link>
          </div>
        </div>
    )
}

export default Error
