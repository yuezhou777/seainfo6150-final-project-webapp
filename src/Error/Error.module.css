.container{
    display: flex;
    flex-flow: column;
    text-align: center;
}

.errorImg {
    margin-bottom: 30px;
}

.link {
    font-family: 'Verdana', sans-serif;
    font-size: 1.5em;
    font-weight: bold;
    color: saddlebrown;
}

@media (min-width: 400px) {
    .board {
        flex-flow: column wrap;
    }
}

@media (min-width: 700px) {
    .board {
        flex-flow: column wrap;
    }
}

@media (min-width: 800px) {
    .board {
        flex-flow: column;
    }
}