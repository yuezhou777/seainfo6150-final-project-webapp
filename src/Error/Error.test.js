import React from 'react'
import {render} from "@testing-library/react";
import Error from "./Error";
import { BrowserRouter } from 'react-router-dom';

it("renders Error Correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Error />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
});