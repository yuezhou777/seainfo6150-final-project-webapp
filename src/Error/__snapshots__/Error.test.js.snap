// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <div
        class="navContainer"
      >
        <header
          class="navContainer"
        >
          <nav>
            <a
              class="navIcon"
              href="/"
            >
              PurrPet
            </a>
          </nav>
          <nav>
            <ul
              class="navList"
            >
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/aboutus"
                >
                  About Us
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/allpets"
                >
                  All Pets
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/category"
                >
                  Categories
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/contactus"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </div>
    <div>
      <img
        alt="404 Not Found"
        class="errorImg"
        src="error.jpg"
      />
    </div>
    <div>
      <a
        class="link"
        href="/"
      >
        Back To Main
      </a>
    </div>
  </div>
</div>
`;

exports[`renders the Error Component Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <div
        class="navContainer"
      >
        <header
          class="navContainer"
        >
          <nav>
            <a
              class="navIcon"
              href="/"
            >
              PurrPet
            </a>
          </nav>
          <nav>
            <ul
              class="navList"
            >
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/aboutus"
                >
                  About Us
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/allpets"
                >
                  All Pets
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/category"
                >
                  Categories
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/contactus"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </div>
    <div>
      <img
        alt="404 Not Found"
        class="errorImg"
        src="error.jpg"
      />
    </div>
    <div>
      <a
        class="link"
        href="/"
      >
        Back To Main
      </a>
    </div>
  </div>
</div>
`;
